llama_index/llms/openai.py

```diff
@@ -0,0 +1,207 @@
+from typing import Any, Awaitable, Dict, Generator, Optional, Sequence
+
+from pydantic import BaseModel, Field
+from llama_index.llm_predictor.base import LLMMetadata
+
+from llama_index.llms.base import (
+    LLM,
+    ChatDeltaResponse,
+    ChatMessage,
+    ChatResponse,
+    ChatResponseType,
+    CompletionDeltaResponse,
+    CompletionResponse,
+    CompletionResponseType,
+    Message,
+)
+from llama_index.llms.generic_utils import (
+    chat_response_to_completion_response,
+    completion_response_to_chat_response,
+    messages_to_prompt,
+    prompt_to_messages,
+)
+from llama_index.llms.openai_utils import (
+    completion_with_retry,
+    is_chat_model,
+    openai_modelname_to_contextsize,
+    to_openai_message_dicts,
+)
+
+
+class OpenAI(LLM, BaseModel):
+    model: str = Field("gpt-3.5-turbo")
+    stream: bool = False
+    temperature: float = 0.0
+    max_tokens: Optional[int] = None
+    additional_kwargs: Dict[str, Any] = Field(default_factory=dict)
+    max_retries: int = 10
```

Comment:
A super common ask is "how do I add a system prompt/prefix/persona to my chatbot?"

Might be a good idea to have some completion/system message support out of the box, rather than asking users to customize the current prompt templates.

Although this has some implications for input prompt size... if it's too annoying it's all good lol

Reply:
Good point. Right now you can configure that when calling the `chat` endpoint, but may be good to have a constructor arg that user can set. that influences all generations.